### PR TITLE
[Feature] Support list partition of cloud native table

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 8 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ListPartitionDesc.java|2|3|66.67%|[242]|
|:large_blue_circle:|com/starrocks/catalog/ListPartitionInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/ListPartitionPersistInfo.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 13 (92.31%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ListPartitionDesc.java|2|3|66.67%|[298]|
|:large_blue_circle:|com/starrocks/load/PartitionUtils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ListPartitionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/ListPartitionPersistInfo.java|1|1|100.00%|[]|

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ListPartitionDesc.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/PartitionUtils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ListPartitionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/ListPartitionPersistInfo.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ListPartitionDesc.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/PartitionUtils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ListPartitionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/ListPartitionPersistInfo.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ListPartitionDesc.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/PartitionUtils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ListPartitionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/ListPartitionPersistInfo.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ListPartitionDesc.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/PartitionUtils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ListPartitionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/ListPartitionPersistInfo.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/ListPartitionInfo.java [hunk 1]
```diff
@@ -455,4 +459,8 @@ public static int compareByValue(List<List<String>> left, List<List<String>> rig
         }
         return 0;
     }
+
+    public void setStorageToCacheInfo(long partitionId, DataCacheInfo dataCacheInfo) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/ListPartitionInfo.java:463]:
```suggestion
    public void setStorageCacheInfo(long partitionId, DataCacheInfo dataCacheInfo) {
```

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ListPartitionDesc.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/PartitionUtils.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ListPartitionInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/ListPartitionPersistInfo.java|1|1|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Comment by TszKitLo40** [comment]
https://github.com/Mergifyio backport branch-3.1

**Pull request merged**